tardis/tests/tests_slow/report.py

```diff
@@ -0,0 +1,174 @@
+"""
+A helper class which works as a plugin to generate the test report and upload it
+to the group server's dokuwiki. It inheirts from the class `HTMLReport` of
+the `pytest-html` plugin. The test report contains the following details:
+
+* The git commit hash on which test run was executed.
+* The time of generation of test report.
+* Number of passes, fails, errors, skips etc.
+* Tabular representation of each method - name, result, duration.
+* Embedded image of plot(s) and error log below a particular method (if any).
+
+As a subclass, this class serves as a plugin and hence, `pytest-html` has to be
+unregistered during the test run for tis plugin to function.
+
+When the integration tests are selected for a particular test run, this class
+is registered as a plugin in `pytest_configure` and subsequently unregistered in
+`pytest_unconfigure`. As a plugin, it implements several "hook" functions
+specified in pytest's official documentation.
+
+
+References
+==========
+1. "Writing Plugins" ( https://pytest.org/latest/writing_plugins.html )
+2. "Hookspec Source" ( https://pytest.org/latest/_modules/_pytest/hookspec.html )
+3. "pytest-html" ( https://www.github.com/davehunt/pytest-html )
+"""
+import datetime
+import pkg_resources
+import os
+import time
+
+# For specifying error while exception handling
+from socket import gaierror
+
+from py.xml import html, raw
+from pytest_html.plugin import HTMLReport
+import tardis
+
+try:
+    import dokuwiki
+except ImportError:
+    dokuwiki = None
```

Comment:
There is no reason to have the `dokuwiki_available` variable. Since we are
ductyping anyway we can catch the case `dokuwiki == None` in the same block.

Wolfgang Kerzendorf <notifications@github.com> schrieb am Do., 9. Juni 2016
um 13:50 Uhr:

> In tardis/tests/tests_slow/report.py
> <https://github.com/tardis-sn/tardis/pull/583#discussion_r66426452>:
>
> > +import datetime
> > +import pkg_resources
> > +import os
> > +import time
> > +
> > +# For specifying error while exception handling
> > +from socket import gaierror
> > +
> > +from py.xml import html, raw
> > +from pytest_html.plugin import HTMLReport
> > +import tardis
> > +
> > +try:
> > +    import dokuwiki
> > +except ImportError:
> > +    dokuwiki = None
>
> why not dokuwiki_available = False
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/tardis-sn/tardis/pull/583/files/39748519f55d18998f52984f9929bfc1876a2782#r66426452>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF4OjvHknV68nNtK4Rc7R-9fGlF0-7Mzks5qJ_4TgaJpZM4Iv3C8>
> .
>